### PR TITLE
Weekly Patch Release v1.3.5 [full merge, no squash]

### PR DESCRIPTION
**Pull request by SeanNaren**
## What does this PR do?

Fixes #<issue_number>

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)
 - [ ] Check that all items from **Before submitting** are resolved
 - [ ] Make sure the title is self-explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,190 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
 
+## [1.4.0] - 2021-MM-DD
```
[the remark this answers is not in the document]
Comment (SeanNaren) [reply CHANGELOG.md]: was working on it, waited for tests to pass

File: CHANGELOG.md [hunk 2]
```diff
@@ -28,6 +48,12 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Fixed training loop total batch counter when accumulate grad batches was enabled ([#7692](https://github.com/PyTorchLightning/pytorch-lightning/pull/7692))
 
 
+- Fixed a bug where checking `trainer.precision` changed to `'mixed'` when specifying 16 in trainer ([#7825](https://github.com/PyTorchLightning/pytorch-lightning/pull/7825))
+
+
+- Fixed `LearningRateMonitor` keys not properly setup when running with `BackboneFinetuning` Callback ([#7835](https://github.com/PyTorchLightning/pytorch-lightning/pull/7835))
+
+
```
Comment (carmocca) [inline CHANGELOG.md:56]: These shouldn't be here?

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,13 +5,33 @@ All notable changes to this project will be documented in this file.
...
+- Added warning to Training Step output ([#7780](https://github.com/PyTorchLightning/pytorch-lightning/pull/7779))
+
+### Fixed
+
+- Fixed LearningRateMonitor + BackboneFinetuning ([#7780](https://github.com/PyTorchLightning/pytorch-lightning/pull/7835))
+- Minor improvements to `apply_to_collection` and type signature of `log_dict` ([#7780](https://github.com/PyTorchLightning/pytorch-lightning/pull/7851))
+- Fixed docker versions ([#7780](https://github.com/PyTorchLightning/pytorch-lightning/pull/7834))
+- Fixed sharded training check for fp16 precision ([#7780](https://github.com/PyTorchLightning/pytorch-lightning/pull/7825))
+- Fixed support for torch Module type hints in LightningCLI ([#7780](https://github.com/PyTorchLightning/pytorch-lightning/pull/7807))
```
Comment (Borda) [inline CHANGELOG.md:20]: everywhere is just 7780

**Review by Borda — CHANGES_REQUESTED** [review]

**Review by awaelchli — APPROVED** [review]
apart from what was mentioned about changelog LGTM :))

**Comment by SeanNaren** [comment]
Thanks for the keen eyes guys, should all be fixed :)

**Review by ethanwharris — APPROVED** [review]

**Review by Borda — CHANGES_REQUESTED** [review]
I am missing #7851 which is mentioned in the chlog

**Pull request merged**